packages/core/content-manager/admin/src/history/components/VersionInputRenderer.tsx

```diff
@@ -218,6 +218,39 @@ type VersionInputRendererProps = DistributiveOmit<EditFieldLayout, 'size'> & {
   shouldIgnoreRBAC?: boolean;
 };
 
+/**
+ * Checks if the i18n plugin added a label action to the field and modifies it
+ * to adapt the wording for the history page.
+ */
+const getLabelAction = (labelAction: VersionInputRendererProps['labelAction']) => {
+  if (!labelAction || typeof labelAction !== 'object' || !('props' in labelAction)) {
+    return labelAction;
+  }
+
+  const customLabelAction =
+    typeof labelAction === 'object' && labelAction != null && 'props' in labelAction
+      ? { ...labelAction, props: { ...labelAction.props } }
+      : labelAction;
+  const labelActionTitleId =
+    labelAction && typeof labelAction === 'object' && labelAction.props.title.id;
+
+  if (labelActionTitleId === 'i18n.Field.localized') {
+    customLabelAction.props.title = {
+      id: 'history.content.localized',
+      defaultMessage:
+        'This value is specific to this locale. If you restore this version, the content will not replaced for other locales.',
+    };
+  } else if (labelActionTitleId === 'i18n.Field.not-localized') {
+    customLabelAction.props.title = {
+      id: 'history.content.not-localized',
+      defaultMessage:
+        'This value is common to all locales. If you restore this version and save the changes, the content will be replaced for all locales.',
+    };
+  }
+
+  return customLabelAction;
+};
```

Comment:
Is `labelAction` a `ReactNode`? IIRC it is, it would be better to do a `cloneElement` rather than accessing the react internals directly... you can also use `isValidElement` as well, since it's not a component reference but at this point we've rendered it too that should work to tell you you can do `cloneElement`. Otherwise i think the solution isn't as hacky as you might think! 😄

Reply:
I'm happy to make the hack less hacky! let me look into this. thanks